spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -196,24 +199,33 @@ private void commitOperation(SnapshotUpdate<?> operation, String description) {
     }
 
     long start = System.currentTimeMillis();
-    operation.commit(); // abort is automatically called if this fails
-    long duration = System.currentTimeMillis() - start;
-    LOG.info("Committed in {} ms", duration);
+    try {
+      operation.commit(); // abort is automatically called if this fails
+      long duration = System.currentTimeMillis() - start;
+      LOG.info("Committed in {} ms", duration);
+    } catch (CommitStateUnknownException commitStateUnknownException) {
```

Comment:
Let's leave this for a follow up then? I'm a little worried about messing more with SnapshotProducer right now.

Reply:
This is our current doc for the underlying commit function
```
   * Implementations must throw a {@link org.apache.iceberg.exceptions.CommitStateUnknownException}
   * in cases where it cannot be determined if the commit succeeded or failed.
   * For example if a network partition causes the confirmation of the commit to be lost,
   * the implementation should throw a CommitStateUnknownException. This is important because downstream users of
   * this API need to know whether they can clean up the commit or not, if the state is unknown then it is not safe
   * to remove any files. All other exceptions will be treated as if the commit has failed.
   ```

Reply:
a separate follow-up sounds good to me

Reply:
It is an interesting idea to interpret all exceptions apart from `CommitFailedException` as commit unknown exceptions but I also worry it is not how the core behaves right now. I'd prefer this approach for now.